src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -990,11 +990,14 @@ def __call__(self, poller):
         except ClientException as client_exception:
             from azure.cli.core.commands.arm import handle_long_running_operation_exception
             self.cli_ctx.get_progress_controller().stop()
+            if getattr(client_exception, 'status_code', None) == 404 and 'delete' in self.cli_ctx.data['command']:
```

Comment:
Even if we add this hack, Python SDK and its users will still suffers from this non-standard behavior.

Reply:
Do we have an open issue in https://github.com/Azure/azure-rest-api-specs to track this?

Reply:
I opened an issue in our repo https://github.com/Azure/azure-cli/issues/15906 for service team to deal with.

Reply:
Based on a disucssion with swagger reviewers, there are some existing services using 404 in LRO. For new swaggers, there should be validation tasks to catch it but for existing ones as long as the json is not modified with PRs, it will not be caught.

Reply:
I still think we should force service team to fix it, instead of patching it from our side, otherwise Python SDK and its users will still suffers from this non-standard behavior.